optimade/validator/validator.py

```diff
@@ -114,7 +114,10 @@ def get(self, request: str):
 
         """
         if request:
-            self.last_request = f"{self.base_url}{request}".replace("\n", "")
+            if any(request.startswith(scheme) for scheme in ("http", "https")):
```

Comment:
Is there a better more general way to test that a string is a valid full URL? If not, fine :)

Reply:
I thought about using `AnyHttpUrl` and validating against that, but I think this is clear enough and it shouldn't be particularly fragile. I guess maybe the comparison should be against `http://` now I think about it.

Reply:
Maybe you could parse it using `urllib.parse.urlparse` and check `scheme` from there? A quick test shows that a random string is passed into the `path` attribute of the `ParseResult`. Then you could simply check if `scheme` is non-empty and of `http` or `https`? I don't know. It may be enough as is as well. I am not fully sure when this method is called, and whether it can be manipulated to be called when not desired?

Reply:
Ah `urllib.parse.urlparse` looks better (if slightly overkill), I didn't expect it to work on the URL fragments.

Reply:
I'm almost certain it's what `AnyHttpUrl` uses in the background anyway?
I am just worried that checks are not robust enough, since I think they should be a bit overkill for the validator, since it should generally be the most robust form of testing an OPTIMADE implementation, in any scenario.

Reply:
Fair enough, I don't think this approach catches the most likely case where it could go wrong, i.e. an implementation providing a next link that doesn't have a URL scheme (which will then trigger the base URL to be prepended to the request). I can't see a way of catching this in general, apart from the validator raising an error (as it does atm) as it has tried to access what should be an obviously incorrect URL.

My last commit adds a few extra calls to urllib where we were doing string manipulation previously.

Reply:
I think raising when it's unable to detect one or another is the correct response, as long as we make sure we _really_ tried ;)